### PR TITLE
add TOML for Gimlet Rev B

### PR DESCRIPTION
**Pull request by bcantrill**
This also renames `gimlet-1` to `gimlet-a`.  Here is the `humility manifest` for `gimlet-a`:

```
     version => hubris build archive v1.0.0
     git rev => 0b79f4805dd8fa6be21f1eaf5740253a63fc1052-dirty
    image id => [3f, e4, ec, be, ab, 4, 14, aa]
       board => gimlet-a
      target => thumbv7em-none-eabihf
    features => itm
  total size => 214K
 kernel size => 23K
       tasks => 13
                ID TASK                SIZE FEATURES
                 0 hiffy              44.0K h753, stm32h7, itm, i2c, gpio, spi, qspi
                 1 gimlet_seq         52.5K h753
                 2 spi4_driver        11.1K spi4, h753
                 3 spi2_driver        11.6K spi2, h753
                 4 i2c_driver         11.8K h753, itm
                 5 spd                19.5K h753, itm
                 6 power              11.5K itm, h753
                 7 hf                  8.2K h753
                 8 jefe                7.0K itm
                 9 thermal             7.6K itm, h753
                10 sensor              4.2K itm
                11 sys                 1.6K h753
                12 idle                0.1K 
   i2c buses => 4 controllers, 5 buses
                C PORT MODE NAME          DESCRIPTION
                1 B    trgt spd           SPD proxy
                2 B    init m2            M.2 bus
                2 F    init front         Front bus
                3 H    init mid           Mid bus
                4 F    init rear          Rear bus
 i2c devices => 26 devices
                C P  MUX ADDR DEVICE        DESCRIPTION
                2 F  -   0x48 tmp117        Front temperature sensor (zone 1)
                2 F  -   0x49 tmp117        Front temperature sensor (zone 2)
                2 F  -   0x4a tmp117        Front temperature sensor (zone 3)
                2 F  -   0x70 pca9545       U.2 ABCD mux
                2 F  -   0x71 pca9545       U.2 EFGH mux
                2 F  -   0x72 pca9545       U.2 IJ mux
                2 B  -   0x73 pca9545       M.2 mux
                3 H  -   0x24 tps546b24a    A2 3.3V rail
                3 H  -   0x27 tps546b24a    A2 1.8V rail
                3 H  -   0x29 tps546b24a    A2 5V rail
                3 H  -   0x3a max5970       M.2 hot plug controller
                3 H  -   0x4c sbtsi         CPU temperature sensor
                3 H  -   0x58 idt8a34003    Clock generator
                3 H  -   0x5a raa229618     CPU power controller
                3 H  -   0x5b raa229618     SoC power controller
                3 H  -   0x5c isl68224      DIMM ABCD power controller
                3 H  -   0x5d isl68224      DIMM EFGH power controller
                4 F  -   0x10 adm1272       Fan hot swap controller
                4 F  -   0x14 adm1272       Sled hot swap controller
                4 F  -   0x20 max31790      Fan controller
                4 F  -   0x48 tmp117        Rear temperature sensor (zone 1)
                4 F  -   0x49 tmp117        Rear temperature sensor (zone 2)
                4 F  -   0x4a tmp117        Rear temperature sensor (zone 3)
                4 F  -   0x4c tmp451        T6 temperature sensor
                4 F  -   0x5f isl68224      T6 power controller
                4 F  -   0x67 bmr491        Intermediate bus converter
```

And here is the manifest for `gimlet-b`:

```
     version => hubris build archive v1.0.0
     git rev => 0b79f4805dd8fa6be21f1eaf5740253a63fc1052-dirty
    image id => [77, 44, fb, f, ad, aa, 84, 1a]
       board => gimlet-b
      target => thumbv7em-none-eabihf
    features => itm
  total size => 214K
 kernel size => 23K
       tasks => 13
                ID TASK                SIZE FEATURES
                 0 hiffy              44.0K h753, stm32h7, itm, i2c, gpio, spi, qspi
                 1 gimlet_seq         52.3K h753
                 2 spi4_driver        11.1K spi4, h753
                 3 spi2_driver        11.6K spi2, h753
                 4 i2c_driver         11.8K h753, itm
                 5 spd                19.5K h753, itm
                 6 power              11.5K itm, h753
                 7 hf                  8.2K h753
                 8 jefe                7.0K itm
                 9 thermal             7.6K itm, h753
                10 sensor              4.2K itm
                11 sys                 1.6K h753
                12 idle                0.1K 
   i2c buses => 4 controllers, 5 buses
                C PORT MODE NAME          DESCRIPTION
                1 B    trgt spd           SPD proxy
                2 B    init m2            M.2 bus
                2 F    init front         Front bus
                3 H    init mid           Mid bus
                4 F    init rear          Rear bus
 i2c devices => 26 devices
                C P  MUX ADDR DEVICE        DESCRIPTION
                2 F  -   0x48 tmp117        Southwest temperature sensor
                2 F  -   0x49 tmp117        South temperature sensor
                2 F  -   0x4a tmp117        Southeast temperature sensor
                2 F  -   0x70 pca9545       U.2 ABCD mux
                2 F  -   0x71 pca9545       U.2 EFGH mux
                2 F  -   0x72 pca9545       U.2 IJ/FRUID mux
                2 B  -   0x73 pca9545       M.2 mux
                2 B  1:3 0x4c tmp451        T6 temperature sensor
                3 H  -   0x24 tps546b24a    A2 3.3V rail
                3 H  -   0x25 tps546b24a    A0 3.3V rail
                3 H  -   0x27 tps546b24a    A2 5V rail
                3 H  -   0x29 tps546b24a    A2 1.8V rail
                3 H  -   0x3a max5970       M.2 hot plug controller
                3 H  -   0x4c sbtsi         CPU temperature sensor
                3 H  -   0x58 idt8a34003    Clock generator
                3 H  -   0x5a raa229618     CPU power controller
                3 H  -   0x5b raa229618     SoC power controller
                3 H  -   0x5c isl68224      DIMM/SP3 1.8V power controller
                4 F  -   0x10 adm1272       Fan hot swap controller
                4 F  -   0x14 adm1272       Sled hot swap controller
                4 F  -   0x20 max31790      Fan controller
                4 F  -   0x25 tps546b24a    T6 power controller
                4 F  -   0x48 tmp117        Northeast temperature sensor
                4 F  -   0x49 tmp117        North temperature sensor
                4 F  -   0x4a tmp117        Northwest temperature sensor
                4 F  -   0x67 bmr491        Intermediate bus converter
```

### CODE REVIEW

File: task/power/src/main.rs [hunk 1]
```diff
@@ -179,6 +180,25 @@ fn controllers() -> [PowerController; 10] {
     ]
 }
 
+#[cfg(target_board = "gimlet-b")]
+fn controllers() -> [PowerController; 11] {
+    let task = I2C.get_task_id();
+
+    [
+        rail_controller!(task, IBC, bmr491, v12_sys_a2, A2),
```
Comment (rmustacc) [inline task/power/src/main.rs:188]: Should the DRAM Vpp ISL68224 be here? This is U352 with outputs for:

* VPP_ABCD_A0_REG
* VPP_EFGH_A0_REG
* V1P8_SP3_VDD_18_A0_REG
* 
In addition, there's the NIC TPS546B24 -- U565; V0P96_NIC_VDD_A0HP.

File: app/gimlet/rev-b.toml [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+bus = "m2"
+address = 0x73
+device = "pca9545"
+description = "M.2 mux"
+refdes = "U422"
+
+[[config.i2c.devices]]
+bus = "m2"
+mux = 1
+segment = 3
```
Comment (rmustacc) [inline app/gimlet/rev-b.toml:339]: I thought segments were 1-based indexes based on what I was specifying to humility. In this case that'd be 4.

File: app/gimlet/rev-b.toml [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+description = "SoC power controller"
+pmbus = { rails = [ "VDDCR_SOC", "VDD_MEM_EFGH" ] }
+sensors = { temperature = 2, power = 2, voltage = 2, current = 2 }
+refdes = "U351"
+
+[[config.i2c.devices]]
+bus = "mid"
+address = 0x5c
+device = "isl68224"
+description = "DIMM/SP3 1.8V power controller"
```
Comment (rmustacc) [inline app/gimlet/rev-b.toml:426]: Mind adding that it's SP3 1.8V A0 power, given that we have an A1 1.8V?

**Review by rmustacc — COMMENTED** [review]
This looks pretty good. I noticed there are a few devices missing from both gimlet-a and gimlet-b. I'd suggest we come back to and add them in a follow up (e.g. not know). This includes:

* Board and Fan FRUIDs
* Potentially adding sensors for M.2/U.2
* Sharkfin MAX5970 / FRUID

File: app/gimlet/rev-b.toml (hunk 2 above)
Comment (bcantrill) [reply app/gimlet/rev-b.toml:339]: Good eyes! I really thought I had doubly checked that, but apparently not!

File: task/power/src/main.rs (hunk 1 above)
Comment (bcantrill) [reply task/power/src/main.rs:188]: Yes -- will add those on the Gimlet A side as well.

File: app/gimlet/rev-b.toml (hunk 3 above)
Comment (bcantrill) [reply app/gimlet/rev-b.toml:426]: Done

File: app/gimlet/rev-b.toml (hunk 2 above)
Comment (bcantrill) [reply app/gimlet/rev-b.toml:339]: Done

File: task/power/src/main.rs (hunk 1 above)
Comment (bcantrill) [reply task/power/src/main.rs:188]: Done

**Pull request merged**